scu_builders.py

```diff
@@ -0,0 +1,335 @@
+"""Functions used to generate scu build source files during build time
+"""
+import glob, os
+import math
+from pathlib import Path
+from os.path import normpath, basename
+
+base_folder = os.path.abspath("./") + "/"
```

Comment:
This is dependant on the folder the script is build from (I guess in theory scons switch to the project root directory before starting the build, but I'm not 100% sure on this)

Instead it's better to use `__file__`:
```suggestion
base_folder = Path(__file__).parent
```